From 3786f98b2a2d882e066897b31bd68bc09bc41126 Mon Sep 17 00:00:00 2001
From: Ron Eldor <Ron.Eldor@arm.com>
Date: Sun, 8 Jul 2018 08:50:31 +0300
Subject: [PATCH 21/43] Fix compilation errors

Fix compilation errors when `MBEDTLS_DTLS_SRTP` not set
1. Add file missed in previous commmit.
2. In sample applications, set `DFL_FORCE_SRTP_PROFILE` to 0.
---
 include/mbedtls/ssl.h      | 16 ++++++++++++++++
 programs/ssl/ssl_client2.c |  2 +-
 programs/ssl/ssl_server2.c |  2 +-
 3 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/include/mbedtls/ssl.h b/include/mbedtls/ssl.h
index e164f2af1..4340a75e0 100644
--- a/include/mbedtls/ssl.h
+++ b/include/mbedtls/ssl.h
@@ -813,6 +813,13 @@ typedef enum
 }
 mbedtls_ssl_srtp_profile;
 
+typedef struct
+{
+    const mbedtls_ssl_srtp_profile   profile;
+    const char                      *name;
+}
+mbedtls_ssl_srtp_profile_info;
+
 typedef struct mbedtls_dtls_srtp_info_t
 {
     mbedtls_ssl_srtp_profile chosen_dtls_srtp_profile; /*!< negotiated SRTP profile */
@@ -2519,6 +2526,15 @@ mbedtls_ssl_srtp_profile mbedtls_ssl_get_dtls_srtp_protection_profile( const mbe
  * \return         0 on succes, MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL if the key buffer is too small to hold the generated key
  */
 int mbedtls_ssl_get_dtls_srtp_key_material( const mbedtls_ssl_context *ssl, unsigned char *key, size_t *key_len );
+
+/**
+ * \brief                  Utility function to get information on dtls srtp profile.
+ *
+ * \param profile          The dtls-srtp profile id to get info on.
+ *
+ * \return         mbedtls_ssl_srtp_profile_info* on success, NULL if not found
+ */
+const mbedtls_ssl_srtp_profile_info *mbedtls_ssl_dtls_srtp_profile_info_from_id( mbedtls_ssl_srtp_profile profile );
 #endif /* MBEDTLS_SSL_DTLS_SRTP */
 
 /**
diff --git a/programs/ssl/ssl_client2.c b/programs/ssl/ssl_client2.c
index 78ed9404b..f07f12ca0 100644
--- a/programs/ssl/ssl_client2.c
+++ b/programs/ssl/ssl_client2.c
@@ -116,7 +116,7 @@ int main( void )
 #define DFL_ETM                 -1
 #define DFL_SKIP_CLOSE_NOTIFY   0
 #define DFL_USE_SRTP            0
-#define DFL_SRTP_FORCE_PROFILE  MBEDTLS_SRTP_UNSET_PROFILE
+#define DFL_SRTP_FORCE_PROFILE  0
 #define DFL_SRTP_MKI            ""
 
 #define GET_REQUEST "GET %s HTTP/1.0\r\nExtra-header: "
diff --git a/programs/ssl/ssl_server2.c b/programs/ssl/ssl_server2.c
index ab34e2eaf..e2d05b89e 100644
--- a/programs/ssl/ssl_server2.c
+++ b/programs/ssl/ssl_server2.c
@@ -160,7 +160,7 @@ int main( void )
 #define DFL_EXTENDED_MS         -1
 #define DFL_ETM                 -1
 #define DFL_USE_SRTP            0
-#define DFL_SRTP_FORCE_PROFILE  MBEDTLS_SRTP_UNSET_PROFILE
+#define DFL_SRTP_FORCE_PROFILE  0
 #define DFL_SRTP_SUPPORT_MKI    0
 
 #define LONG_RESPONSE "<p>01-blah-blah-blah-blah-blah-blah-blah-blah-blah\r\n" \
-- 
2.17.1

